README.md

```diff
@@ -1,2 +1,211 @@
+
+## Installation for plugin developers
+
+Run `jspm i aurelia-config` from your plugin root resp. `npm i aurelia-config --save`.
+
+## Installation for applications
+
+### Aureli-Cli
+
+Run `npm i aurelia-config --save` from your project root.
+
+Aurelia-config makes use [homefront](https://www.npmjs.com/package/homefront). So, add following to the `build.bundles.dependencies` section of `aurelia-project/aurelia.json`.
+
+```js
+"dependencies": [
+  // ...
+  {
+    "name": "homefront",
+    "path": "../node_modules/homefront/dist",
+    "main": "index"
+  },
+  "aurelia-config",
+  // ...
+],
+```
+
+### Jspm
+
+Run `jspm i aurelia-config` from your project root.
+
+Add `aurelia-config` to the `bundles.dist.aurelia.includes` section of `build/bundles.js`.
+
+If the installation results in having forks, try resolving them by running:
+
+```sh
+jspm inspect --forks
+jspm resolve --only registry:package-name@version
+```
+
+### Webpack
+
+Run `npm i aurelia-config --save` from your project root.
+
+Add `'aurelia-config'` in the `coreBundles.aurelia section` of your `webpack.config.js`.
+
+### Typescript
+
+Npm-based installations pick up the typings automatically. For Jspm-based installations, run `typings i github:spoonx/aurelia-config` or add `"aurelia-config": "github:spoonx/aurelia-config",` to your `typings.json` and run `typings i`.
+
+## Usage
+
+### Configure aurelia-config in your app
+
+Your basic configuration file just exports a simple object with (preferably name-spaced) configuration settings, eg.:
+
+```js
+export let appConfig = {
+  'aurelia-api': {
+    api: 'http://example.com/api',
+    auth: 'http://example.com/auth'
+  },
+  'aurelia-authentication': {
+    endpoint: 'auth',
+    configureEndpoints: true
+  }
+}
+```
+
+Register the with aurelia-config the plugin configs and your appConfig.
+
+```js
+let someConfig = { key: 'value'};
+
+aurelia.use
+  // aurelia-config should be before other plugins which use aurelia-config so that they can use your appConfig setting already during configuration
+  .plugin('aurelia-config', [
+    /* a plugin which follows the convention aka exports a object named 'config' */
+    'aurelia-api',
+    /* a plugin which exports a configuration object named 'baseConfig' */
+    {
+      name: 'aurelia-authentication',
+      configName: 'baseConfig'
+    },
+    /* a configuration object */
+    someConfig,
+    /* a local configuration which which exports a object named 'config' */
+    './appConfig'
+  ])
+  /* Other plugins that use aurelia-config don't need any or much configuration anymore */
+  .plugin('aurelia-api')
+  /* but you still can overwrite or add settings if the plugin allows that*/
+  .plugin('aurelia-authentication', {
+    configureEndpoints: 'false'
+  });
+```
+
+Configs get added in order of the array. So, later called plugins overwrite previous settings. Hence, your `appConfig` should be the last entry in that array.
```

Comment:
I'd explain this differently. The concept of a global object isn't clear for people new to this. I'd put this in the book :)